File: t/ruby/EntryExit.rb

require 'TestClass'

obj = Sm::TestClass::new
obj.getFSM().setDebugStream($stdout)
obj.getFSM().setDebugFlag(ARGV.size > 0)
obj.Evt_1()
obj.Evt_2()
obj.Evt_3()
obj.Evt_1()
obj.Evt_2()
obj.Evt_3()

File: t/javascript/EntryExit.js

require('./Sm/TestClass.js');

var obj = new TestClass();
obj.fsm.setDebugStream(process.stdout);
obj.fsm.setDebugFlag(process.argv.length > 2);
obj.Evt_1();
obj.Evt_2();
obj.Evt_3();
obj.Evt_1();
obj.Evt_2();
obj.Evt_3();
